### PR TITLE
Block too many concurrent mapping updates

### LINKED ISSUE #50670: Dynamic mapping updates are unboundedly parallel
Before 7.2.0 dynamic mapping updates would block a `write` thread waiting for the master to acknowledge the new mapping. In #39793 we moved to an asynchronous model, freeing up the `write` thread to carry on with other indexing tasks.

One feature of the pre-7.2.0 blocking approach was that the number of `write` threads is limited and this limits the number of parallel dynamic mapping updates pending on the master. The asynchronous model has no such limit and may send a very large number of dynamic mapping updates in a short time since the shard bulks are processed much faster. Furthermore, many indexing operations may require the same mapping update but since they are now generated much more quickly it may be that many of the mapping updates sent to the master are duplicates.

Related [discussion thread](https://discuss.elastic.co/t/performance-degradation-after-upgrading-from-v6-3-2-to-v7-4-2/213466/5?u=davidturner).

### PR DESCRIPTION
**Pull request by ywelsch**
Ensures that there are not too many concurrent dynamic mapping updates going out from the data nodes to the master.

Closes https://github.com/elastic/elasticsearch/issues/50670

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java]: Sadly, this changes exception handling to renotify onFailure if onResponse fails. Not sure if it is important here, but I would opt to keep the explicit listener until #50886 is merged.

File: server/src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java]: nit: I think this comment is unnecessary (but feel free to leave it in)

File: server/src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java]: nit: I think this comment is unnecessary (but feel free to leave it in)

**Review by henningandersen — APPROVED** [review]
LGTM (provided tests succeed).

Left a couple of optional comments.

**Comment by ShauliSolomovich** [comment]
Hi, 
i saw this fix in the release notes of version `7.5.2 & 7.6.0`, 
but this code is still missing on those releases and also in `7.6.2`,
any plan to add it to the next release ?

**Pull request merged**
